indico/modules/users/models/users.py

```diff
@@ -396,3 +443,45 @@ def unlink_to(self, obj, role):
         :param role: the role to use in the link
         """
         return UserLink.remove_link(self, obj, role)
+
+    def synchronize_data(self, refresh=False):
+        """
+        Synchronize the fields of the user from the sync identity.
+
+        This will take only into account ``user.sync_fields``.
+
+        :param refresh: bool -- Whether to refresh the synced identity
+                        with the sync provider before or not. (Only if
+                        the sync provider support refresh.)
+        """
+        identity = self._get_synced_identity(refresh=refresh)
+        if identity is None:
+            return
+        for field in self.synced_fields:
+            old_value = getattr(self, field)
+            new_value = identity.data.get(field) or ''
+            if field in ('first_name', 'last_name') and not new_value:
+                continue
+            if old_value == new_value:
+                continue
+            flash(_("Your {field_name} has been synchronised from '{old_value}' to '{new_value}'.").format(
```

Comment:
I agree with the ugliness of multiple flash messages. A nice fix would be to have a simple way to build a multilines flash message with the list of fields which were synchronized.
I the mean time, I think having a few flash messages (worst case is 5) is good enough.